packages/insomnia-scripting-environment/src/objects/send-request.ts

```diff
@@ -15,35 +15,28 @@ export async function sendRequest(
   cb: (error?: string, response?: Response) => void,
   settings: Settings,
 ): Promise<Response | undefined> {
-  return new Promise<Response | undefined>(async resolve => {
-    // TODO(george): enable cascading cancellation later as current solution just adds complexity
-    const requestOptions = requestToCurlOptions(request, settings);
-
+  return new Promise(async (resolve, reject) => {
     try {
+      const requestOptions = requestToCurlOptions(request, settings);
       const nodejsCurlRequest =
         process.type === 'renderer'
           ? window.bridge.curlRequest
           : (await import('insomnia/src/main/network/libcurl-promise')).curlRequest;
-      nodejsCurlRequest(requestOptions)
-        .then((result: any) => {
-          const output = result as CurlRequestOutput;
-          return curlOutputToResponse(output, request);
-        })
-        .then((transformedOutput: Response) => {
-          cb(undefined, transformedOutput);
-          resolve(transformedOutput);
-        })
-        .catch((e: string | undefined) => {
-          cb(e, undefined);
-          resolve(undefined);
-        });
-    } catch (err: any) {
-      if (err.name === 'AbortError') {
-        cb(`Request was cancelled: ${err.message}`, undefined);
+
+      const output = (await nodejsCurlRequest(requestOptions)) as CurlRequestOutput;
+      const transformedOutput = await curlOutputToResponse(output, request);
+
+      if (cb) {
+        cb(undefined, transformedOutput);
+      }
+      return resolve(transformedOutput);
+    } catch (e) {
+      if (cb) {
+        cb(e, undefined);
+        resolve(undefined);
       } else {
-        cb(`Something went wrong: ${err.message}`, undefined);
+        reject(e);
       }
-      resolve(undefined);
     }
   });
```

Comment:
Nit: Just realized that, this is an async function, so we could directly remove this promise and use return/throw to replace the resolve/reject.

```suggestion
    try {
      const requestOptions = requestToCurlOptions(request, settings);
      const nodejsCurlRequest =
        process.type === 'renderer'
          ? window.bridge.curlRequest
          : (await import('insomnia/src/main/network/libcurl-promise')).curlRequest;

      const output = (await nodejsCurlRequest(requestOptions)) as CurlRequestOutput;
      const transformedOutput = await curlOutputToResponse(output, request);

      if (cb) {
        cb(undefined, transformedOutput);
      }
      return transformedOutput;
    } catch (e) {
      if (cb) {
        cb(e, undefined);
      } else {
        throw e;
      }
    }
```

Reply:
Aligned offline with @ihexxa, thanks for the details explanation. 👍